### PR TITLE
Edpub 1394 edit section

### PR DESCRIPTION
**Pull request by ckd002**
# Description

Adds an API endpoint for editing a section

## Spec

See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1394

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Pull related branches locally.
3. Use the swagger page to POST to the /api/data/sections/{id} endpoint and verify that you get the expected return.

---

## Change Log
* Adds endpoint for editing a section

### CODE REVIEW

File: src/nodejs/lambda-layers/database-util/src/query/section.js
Comment (camposeddie) [inline src/nodejs/lambda-layers/database-util/src/query/section.js:67]: What are your thoughts on removing the explicit default declarations? For example:
```
    table: `section (${params.id ? 'id,': ''} form_id, heading, list_order${params.required_if ? ', required_if' : ''}${params.show_id ? ', show_if' : ''}${params.daac_id ? ', daac_id)' : ''}`,
    values: {
      type: 'values_list',
      items: [
        ...(params.id ? ['{{id}}'] : []),
        '{{form_id}}', '{{heading}}', '{{list_order}}',
        ...(params.required_if ? `'${JSON.stringify(params.required_if)}'::JSONB` : []),
        ...(params.show_if ? [`'${JSON.stringify(params.show_if)}'::JSONB`] : []),
        ...(params.daac_id? '{{daac_id}}': []),
```
Mainly trying to avoid any confusions around the db defaults versus explicit defaults we add.

Comment (ckd002) [reply src/nodejs/lambda-layers/database-util/src/query/section.js:67]: I waffled on doing it originally so happy to make the change. Only reason I didn't was in someways it makes it harder to read.

File: src/openapi/openapi.json
Comment (camposeddie) [reply src/openapi/openapi.json:12988]: Digging more into this today, and it just gets more confusing 😂 This configuration is erroring as if the mapping is not being accepted; however, if I don't touch the configuration and just deploy the API stage in APIGateway, it starts working fine. 🤷 This makes me think it has something to do with how the configuration is being converted from this input JSON string to the final value within the mapping template.

Comment (camposeddie) [reply src/openapi/openapi.json]: Still a bit stuck on this. I'm going to have to revisit again later so as not to block merging other PRs. I don't think it has anything directly to do with the code written; however, I want to be sure that the issue doesn't occur in higher environments once merged.

Comment (camposeddie) [reply src/openapi/openapi.json]: This issue should be resolved now. The cause was that the permission being granted between the APIGateway and Lambda were being restricted to just the `GET` request method. This was an issue in a couple other lambda permissions as well. The reason the cause wasn't as obvious and why saving manually within AWS seemed to solve the issue is because AWS was implicitly adding those needed permissions when manually updated in the AWS console.

**Comment by camposeddie** [comment]
Validated in SIT with the following commands:
```
INSERT INTO section VALUES ('fc708012-e707-4c37-95a8-cd76351b8f27', '6c544723-241c-4896-a38c-adbc0a364293', 'Testing Edit Section', 1000, '[]', '[]', NULL);
curl -XPOST "https://pub.sit.earthdata.nasa.gov/api/data/sections/fc708012-e707-4c37-95a8-cd76351b8f27" \
    -H 'Authorization: <token>' \
    -d '{"id":"fc708012-e707-4c37-95a8-cd76351b8f27","form_id":"6c544723-241c-4896-a38c-adbc0a364293","heading":"Another Updated Testing Edit Section","list_order":3000,"required_if":[],"show_if":[],"daac_id":"40397fe8-4841-4e4c-b84a-6ece359ff5ff"}'
DELETE FROM section WHERE id='fc708012-e707-4c37-95a8-cd76351b8f27';
```

**Review by camposeddie — APPROVED** [review]

**Pull request merged**
